client/app/components/form/MultiStepFormBase.tsx

```diff
@@ -61,10 +64,23 @@ const MultiStepFormBase = ({
   };
 
   const isDisabled = disabled || isSubmitting;
+  const isCustomTitles = customHeaderTitles && customHeaderTitles.length > 0;
+
+  if (
+    isCustomTitles &&
+    formSectionTitles.length !== customHeaderTitles.length
+  ) {
+    throw new Error(
+      "The number of custom header titles must match the number of form sections",
+    );
+  }
 
   return (
     <>
-      <MultiStepHeader step={formSection} steps={formSectionTitles} />
+      <MultiStepHeader
+        step={formSection}
+        steps={isCustomTitles ? customHeaderTitles : formSectionTitles}
+      />
```

Comment:
Hmmm well I don't buy that it's somehow hard to follow but I don't feel strongly about it and have changed the prop to `customStepNames`.

Reply:
The prop name helps, thanks. In the future when responding to review comments, could you please tone down the skepticism? You've got a lot of experience with RJSF and you wrote this code, so it makes sense that it's going to be easier for you to follow than for someone else. I make hard-to-follow comments whenever it takes me a long time to review a PR relative to how big it is.

Reply:
I'm so sorry, I am just not a huge fan of naming nits as what makes perfect sense to one person may not to another and in most libraries you need to double check what certain props actually do. Though I promise I will be better and more receptive in the future 🙇 

To expand on this a bit I think my main issue with this was the comment about not using `title` since RJSF uses it. Though to me using `custom` ... `title` in what is an RJSF component, replacing RJSF titles with custom ones is fine. Though I think `customStepNames` is nice and clear and I'm happy with the change.